//
// Copyright 2020 The Project Oak Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package oak.identity;

// Response from the client to a signing challenge. The client uses its private key to sign the hash
// of the challenge obtained from the server. The response contains the signed hash along with the
// public key corresponding to the private key used for signing.
message SignedChallenge {
  bytes signed_hash = 1;

  // Public key as a binary DER-encoded `SubjectPublicKeyInfo` (see
  // https://tools.ietf.org/html/rfc5280#section-4.1).
  bytes public_key = 2;
}
